Add an external link icon and css style.

StevenDufresne opened the pull request:
External links should display an external link icon. This PR adds an `svg`, taken from Gutenberg, and a class to target external links.

I wasn't exactly sure what color to give the icon. I went with a moderately dark grey color from our theme. 

<!-- Reference any related issues or PRs here. Each issue needs the "fixes" keyword if the PR fixes more than one thing. -->
See https://github.com/WordPress/wporg-showcase-2022/issues/42.


### Drawback of this approach
 Since we are only using CSS and not a block or something similar, users cannot see the external icon when viewing the post in the block editor. However, I don't expect there to be many external links, and doing more work should probably be considered only if we receive feedback from teams (maybe the documentation team may find that frustrating for example). 

### Screenshots

<!-- If your change has a visual component, add a screenshot here. -->

| Caption | Screenshot | 
| --- | --- |
|  Default | <img width="200" alt="Screen Shot 2022-11-15 at 9 55 59 AM" src="https://user-images.githubusercontent.com/1657336/201801902-4f8bd782-ac54-48a9-bcfd-33bcd3943300.png"> | 
|  With Element font-size: 40px | <img width="400" alt="Screen Shot 2022-11-15 at 9 55 47 AM" src="https://user-images.githubusercontent.com/1657336/201801908-9538f0e3-234f-471c-94bf-2411ae4b8abe.png"> | 
|  Inline | <img width="662" alt="Screen Shot 2022-11-15 at 10 45 32 AM" src="https://user-images.githubusercontent.com/1657336/201806247-2846a192-1331-4d28-bcdd-b01f84a4e849.png"> | 
|  Alternative color option -- Not Implemented here | <img width="200" alt="Screen Shot 2022-11-15 at 9 56 41 AM" src="https://user-images.githubusercontent.com/1657336/201801898-7e6d71d4-df6a-4223-939a-7b7684a2b9b0.png"> | 



### How to test the changes in this Pull Request:

- Add a link to some post content, add `external-link` class to the element. Expect to the the icon.
- Alternatively, you can add the class to the direct parent of the `<a>` 
  - This is done in case we don't have direct access to the `<a>` element.

ryelle commented on source/wp-content/themes/wporg-parent-2021/sass/base/_elements.scss:
```diff
@@ -28,6 +28,22 @@ mark.has-inline-color {
 }
 
+// For links that navigate users to different site
+.external-link > a::after, // In case we don't access to <a> tag
+a.external-link::after {
```
Instead of a custom class, could you use attribute selection to check if the `href` does not contain `wordpress.org`? I think this would work `a:not([href*="wordpress.org"])` ?

StevenDufresne replied:
Yep, this appears to work on showcase (and a few other tests).

StevenDufresne replied:
It's important to note that this is aggressive :)

**Footer**
<img width="381" alt="Screen Shot 2022-11-17 at 12 35 33 PM" src="https://user-images.githubusercontent.com/1657336/202348792-b544e395-8695-4532-bfa2-c0e3863c9ebb.png">

I'll need to run a more in-depth test before merging. Switching this back to a draft until I can.

StevenDufresne replied:
After testing this patch using `wporg-main`, I think I'm tempted to move back to having an `external` class. I think we'll end up with a lot of places that have external URLs that we don't want to have the icon. 

<img width="486" alt="Screen Shot 2022-11-22 at 3 59 11 PM" src="https://user-images.githubusercontent.com/1657336/203248353-dc85d5ae-7df2-4fb7-a693-2dda1d55757a.png">
<img width="557" alt="Screen Shot 2022-11-22 at 4 16 38 PM" src="https://user-images.githubusercontent.com/1657336/203249470-e889a119-d519-4f91-bb5f-9a6b8913fe55.png">

ryelle replied:
Fair enough… unfortunately there's no way to add a class to a link in the editor (there is [this issue](https://github.com/WordPress/gutenberg/issues/23011#issuecomment-1047766085) but I doubt anyone is actively working on it).

I think you'll need some kind of `has-external-link` class that can be added via Advanced (no need to make a block style), but then also combine it with the `:not(…)` code to avoid accidentally marking internal links if there are multiple in a paragraph, for example.

StevenDufresne replied:
This is why I added a second `CSS` line to the declaration, `.external-link > a::after,` so it can be added to the parent. Not super excited about that but at least it works?

https://github.com/WordPress/wporg-parent-2021/pull/53/commits/1723230ccce77a0a8cbfdb2c41c4e1fc41c4923e#diff-dc9ae95ab0a8f0d1179f51200575b6479f49df7f5517f1afeef40a86aa475d09R32

StevenDufresne replied:
> I think you'll need some kind of has-external-link class that can be added via Advanced (no need to make a block style), but then also combine it with the :not(…) code to avoid accidentally marking internal links if there are multiple in a paragraph, for example.

Ah originally misread that. Yeah, multiple links is an issue. Thanks for flagging that.

ryelle commented on source/wp-content/themes/wporg-parent-2021/sass/base/_elements.scss:
```diff
@@ -28,6 +28,21 @@ mark.has-inline-color {
+	height: 1em;
+	margin-left: 0.1em;
+	vertical-align: middle;
+	-webkit-mask-image: url(../images/external-link-icon.svg);
+	mask-image: url(../images/external-link-icon.svg);
+	-webkit-mask-size: cover;
```
The build process here has autoprefixer, so that should handle adding in the `-webkit-` properties.

StevenDufresne commented:
Here's the updated view:

| Place | Image | 
| --- | --- |
| Regular link | <img width="206" alt="Screen Shot 2022-12-08 at 4 19 03 PM" src="https://user-images.githubusercontent.com/1657336/206384445-0b6abd93-2e67-4bd7-9ef5-8dc698927f9a.png"> |
| Scales with font size | <img width="633" alt="Screen Shot 2022-12-08 at 4 24 13 PM" src="https://user-images.githubusercontent.com/1657336/206384571-2297e161-f2b3-4373-9cbb-81d4be2d0936.png"> |
| Embedded link | <img width="809" alt="Screen Shot 2022-12-08 at 4 18 35 PM" src="https://user-images.githubusercontent.com/1657336/206384457-06a2f641-4c01-4255-af74-7a8673d5e9dd.png"> |
| Embedded link `light` | <img width="812" alt="Screen Shot 2022-12-08 at 4 18 26 PM" src="https://user-images.githubusercontent.com/1657336/206384696-47cb2eff-927b-45ce-ab96-0c67db75ec62.png"> |

**Notes:**
- I have added a `external-link-light` for places where we have dark backgrounds
- I have also added styles in case we have access to the anchor tag directly

ryelle reviewed (approved):
Looks good 👍🏻
